src/control/KinematicSimulationTrajectoryExecutor.cpp

```diff
@@ -30,19 +31,24 @@ KinematicSimulationTrajectoryExecutor::KinematicSimulationTrajectoryExecutor(
 KinematicSimulationTrajectoryExecutor::~KinematicSimulationTrajectoryExecutor()
 {
   {
-    std::lock_guard<std::mutex> lockSpin(mMutex);
-    mInExecution = false;
+    std::lock_guard<std::mutex> lock(mMutex);
+    DART_UNUSED(lock); // Suppress unused variable warning
+
+    mInProgress = false;
   }
 }
 
 //==============================================================================
-std::future<void> KinematicSimulationTrajectoryExecutor::execute(
+void KinematicSimulationTrajectoryExecutor::validate(
     trajectory::TrajectoryPtr traj)
 {
   if (!traj)
     throw std::invalid_argument("Traj is null.");
 
-  auto space = std::dynamic_pointer_cast<MetaSkeletonStateSpace>(
+  if (traj->metadata.executorValidated)
+    return;
```

Comment:
What's the purpose of this check? Is there any harm in doing the validation again?

Reply:
This is how I'm attempting to avoid the deadlock issue I mentioned here: https://github.com/personalrobotics/aikido/pull/259#issuecomment-345492616.

In `libherb`, `Herb::step` acquires the lock on the robot skeleton before calling `QueuedTrajectoryExecutor::step`. If we re-validate, then we'll attempt to acquire the lock again which will cause a deadlock. I think a (dangerous?) alternative is to not lock the skeleton while doing this check.